packages/dictionary/src/metaSchema/restrictionsSchemas.ts

```diff
@@ -111,4 +114,99 @@ export const RestrictionRegex = zod.string().superRefine((data, context) => {
+export const StringArrayFieldValue = StringFieldValue.array();
+export const NumberFieldValue = zod.number();
+export const NumberArrayFieldValue = NumberFieldValue.array();
+export const IntegerFieldValue = zod.number().int();
+export const IntegerArrayFieldValue = IntegerFieldValue.array();
+export const BooleanFieldValue = zod.boolean();
+export const BooleanArrayFieldValue = BooleanFieldValue.array();
+
+export const FieldValue = zod.union([
+	BooleanFieldValue,
+	BooleanArrayFieldValue,
+	IntegerFieldValue,
+	IntegerArrayFieldValue,
+	NumberFieldValue,
+	NumberArrayFieldValue,
+	StringFieldValue,
+	StringArrayFieldValue,
+]);
+export type FieldValue = zod.infer<typeof FieldValue>;
+
+export const ArrayTestCase = zod.enum(['all', 'any', 'none']);
+export type ArrayTestCase = zod.infer<typeof ArrayTestCase>;
+
+export const ARRAY_TEST_CASE_DEFAULT = ArrayTestCase.Values.all;
+
+export const MatchRuleCodeList = RestrictionCodeList;
+export type MatchRuleCodeList = zod.infer<typeof MatchRuleCodeList>;
+
+export const MatchRuleCount = zod.number().or(RestrictionIntegerRange);
+export type MatchRuleCount = zod.infer<typeof MatchRuleCount>;
+
+export const MatchRuleExists = zod.boolean();
+export type MatchRuleExists = zod.infer<typeof MatchRuleExists>;
+
+export const MatchRuleRange = RestrictionRange;
+export type MatchRuleRange = zod.infer<typeof MatchRuleRange>;
+
+export const MatchRuleRegex = RestrictionRegex;
+export type MatchRuleRegex = zod.infer<typeof MatchRuleRegex>;
+
+export const MatchRuleValue = FieldValue;
+export type MatchRuleValue = zod.infer<typeof MatchRuleValue>;
+
+export const MatchRule = zod.union([
+	MatchRuleCodeList,
+	MatchRuleCount,
+	MatchRuleExists,
+	MatchRuleRange,
+	MatchRuleRegex,
+	MatchRuleValue,
+]);
+export type MatchRule = zod.infer<typeof MatchRule>;
+
+export const ConditionMatch = zod
+	.object({
+		codeList: MatchRuleCodeList,
+		count: MatchRuleCount,
+		exists: MatchRuleExists,
+		range: MatchRuleRange,
+		regex: MatchRuleRegex,
+		value: MatchRuleValue,
+	})
+	.partial();
+type ConditionMatch = zod.infer<typeof ConditionMatch>;
+
+export const RestrictionCondition = zod.object({
+	fields: zod.string().array(),
+	match: ConditionMatch,
+	case: ArrayTestCase.optional(),
+});
+export type RestrictionCondition = zod.infer<typeof RestrictionCondition>;
+
+export const ConditionalRestrictionTest = zod.object({
+	conditions: zod.array(RestrictionCondition),
+	case: ArrayTestCase.optional(),
+});
+export type ConditionalRestrictionTest = zod.infer<typeof ConditionalRestrictionTest>;
+
+export type ConditionalRestriction<TRestrictionObject> = {
+	if: ConditionalRestrictionTest;
+	then?:
+		| TRestrictionObject
+		| ConditionalRestriction<TRestrictionObject>
+		| (TRestrictionObject | ConditionalRestriction<TRestrictionObject>)[];
+	else?:
+		| TRestrictionObject
+		| ConditionalRestriction<TRestrictionObject>
+		| (TRestrictionObject | ConditionalRestriction<TRestrictionObject>)[];
+};
```

Comment:
This is the shape of a ConditionalRestriction which all the specific implementations must adhere to.